app/services/event_service.rb

```diff
@@ -0,0 +1,48 @@
+# typed: strict
+# frozen_string_literal: true
+
+# Triggers state changes, logs events, and broadcasts changes
+class EventService
+  extend T::Sig
+
+  class << self
+    include ActionView::Helpers::UrlHelper
+  end
+
+  sig { params(work: Work, user: User, description: String).returns(T.nilable(Integer)) }
+  def self.reject(work:, user:, description:)
```

Comment:
Because the methods in this class are so similar, I was tempted to refactor it but I could either way.